src/context/json.ts

```diff
@@ -7,12 +16,27 @@ import { ResponseTransformer } from '../response'
  * res(json('Some string'))
  * res(json([1, '2', false, { ok: true }]))
  */
-export const json = <BodyType>(
-  body: BodyType,
-): ResponseTransformer<BodyType> => {
+export const json = <BodyTypeJSON, BodyTypeString extends string>(
+  body: BodyTypeJSON,
+  {
+    merge = defaultJSONContextOptions.merge,
+  }: JSONContextOptions = defaultJSONContextOptions,
+): ResponseTransformer<BodyTypeString> => {
   return (res) => {
     res.headers.set('Content-Type', 'application/json')
-    res.body = JSON.stringify(body) as any
+
+    if (merge) {
+      try {
+        const nextBody = JSON.parse(res.body)
+
+        res.body = JSON.stringify(mergeRight(body, nextBody)) as BodyTypeString
```

Comment:
That sounds nice @kettanaito can you maybe point me where the `res() chain itself` resides and give me a hint please. I struggle to find my way around.

Reply:
Of course. Here's where the `res` is defined:

https://github.com/mswjs/msw/blob/2b133436cd405a9ebf6d6882d5b8e47997b943f6/src/response.ts#L40-L56

The `resolvedResponse` is the final instance of response composed by it and send to the worker. I think we can stringify the response body before the return statement. Just pay attention if this is a safe thing to do for all `Content-Type` of responses (i.e. should we stringify body for `text/plain` response?).

Reply:
Got ya, cheers. I think we should start with just `JSON.stringifying` the JSON Content-type. This will also solve the issue of `json` returning a `string` which caused some typing annoyances (e.g. the old `any` type). Will look into it soon.

Reply:
Sounds reasonable. Keep in mind there are multiple JSON content-types, we may check them as something like `res.headers.get('content-type')?.endsWith('json')`.

Reply:
Here would be my first propsal: 7bf236d. I will comment also inline.